### PR TITLE
Add deduplication to data models

### LINKED ISSUE #62: Prevent duplicate data from being displayed
- e.g. if some data gets loaded 2x, it shouldn't be displayed 2x, because when adding it via the data service, the id should be recognized and marked as added, but not actually be added to the map

- This can probably be accomplished by having a global map of `{[s:id]: ResponseObject}`

### PR DESCRIPTION
**Pull request by vikaspotluri123**
- closes #62

### CODE REVIEW

File: src/app/services/deduplication.service.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export default new Map<string, true>();
```
Comment (joshcosta) [inline src/app/services/deduplication.service.ts]: Think you could use a Set here since the boolean isn't getting used for anything.

Comment (vikaspotluri123) [reply src/app/services/deduplication.service.ts]: O(n) vs O(1) 😄

Comment (vikaspotluri123) [reply src/app/services/deduplication.service.ts]: O(n) vs O(1) 😄

Comment (vikaspotluri123) [reply src/app/services/deduplication.service.ts]: I sit corrected

**Pull request merged**